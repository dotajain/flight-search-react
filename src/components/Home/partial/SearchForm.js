import React from 'react';
import PropTypes from 'prop-types';

import moment from 'moment';

import { connect } from 'react-redux';

import {
  handleReturnFlight,
  handleOneWayFlight,
  handleSearch,
} from '../../../actions';

import SearchFilter from './SearchFilter';
import TabButton from './TabButton';
import AutoSuggestInput from './AutoSuggestInput';

const SearchForm = ({
  handleReturnFlight,
  handleOneWayFlight,
  flightSearch,
  handleSearch,
}) => {
  const minDate = moment().format('YYYY-MM-DD');

  let returnForm;
  if (flightSearch.returnFlight) {
    returnForm = (
      <div className="form__field form__field--date">
        <label htmlFor="returnDate" className="form__label">
          Return
        </label>
        <div className="form__field--contents">
          <input type="date" name="returnDate" id="returnDate" min={minDate} />
        </div>
      </div>
    );
  }

  const onSearch = e => {
    e.preventDefault();

    const {
      origin: { value: origin },
      destination: { value: destination },
      departureDate: { value: departureDate },
      passengers: { value: passengers },
    } = e.target;

    let returnDate;
    if (e.target.returnDate) {
      returnDate = e.target.returnDate.value;
    }

    handleSearch({
      origin,
      destination,
      departureDate,
      returnDate,
      passengers,
    });
  };

  return (
    <div className="col col--3 search__form">
      <div className="box">
        <div className="tab box__tab">
          <TabButton
            onClick={handleOneWayFlight}
            label="One Way"
            active={!returnForm}
          />
          <TabButton
            onClick={handleReturnFlight}
            label="Return"
            active={!!returnForm}
          />
        </div>

        <div className="box__card box__card--tab">
          <form className="form" onSubmit={onSearch}>
            <AutoSuggestInput
              name="origin"
              id="origin"
              placeholder="Enter Origin City"
              label="Origin City"
            />

            <AutoSuggestInput
              name="destination"
              id="destination"
              placeholder="Enter Destination City"
              label="Destination City"
            />

            <div className="date__picker">
              <div className="form__field form__field--date">
                <label htmlFor="departureDate" className="form__label">
                  Departure
                </label>
                <div className="form__field--contents">
                  <input
                    type="date"
                    name="departureDate"
                    id="departureDate"
                    min={minDate}
                    defaultValue={minDate}
                  />
                </div>
              </div>
              {returnForm}
            </div>
            <div className="form__field form__field--select">
              <label htmlFor="passengers" className="form__label">
                Passangers
              </label>
              <div className="form__field--contents">
                <select name="passengers" id="passangers">
                  <option value="1">1</option>
                  <option value="2">2</option>
                  <option value="3">3</option>
                  <option value="4">4</option>
                  <option value="5">5</option>
                </select>
              </div>
            </div>
            <div className="form__btn--pannel">
              <button>Search</button>
            </div>
          </form>
        </div>
      </div>
      <SearchFilter />
    </div>
  );
};

SearchForm.propTypes = {
  handleOneWayFlight: PropTypes.func,
  handleReturnFlight: PropTypes.func,
  handleSearch: PropTypes.func,
  flightSearch: PropTypes.object,
};

const mapStateToProps = state => ({ flightSearch: state.flightSearch });
export default connect(mapStateToProps, {
  handleReturnFlight,
  handleOneWayFlight,
  handleSearch,
})(SearchForm);
