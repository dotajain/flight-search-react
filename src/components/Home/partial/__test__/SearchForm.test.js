import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../store';

import SearchForm from '../SearchForm';

global.fetch = jest.fn(
  () =>
    new Promise(resolve => {
      process.nextTick(() => resolve({ json: () => ({}) }));
    }),
);

it('Properly rendered SearchForm Component', () => {
  const component = renderer.create(<Provider store={store}><SearchForm /></Provider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
