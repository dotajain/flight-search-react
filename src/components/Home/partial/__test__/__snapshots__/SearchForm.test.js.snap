// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properly rendered SearchForm Component 1`] = `
<div
  className="col col--3 search__form"
>
  <div
    className="box"
  >
    <div
      className="tab box__tab"
    >
      <button
        className="tab__nav tab__nav--active"
        onClick={[Function]}
        type="button"
      >
        One Way
      </button>
      <button
        className="tab__nav "
        onClick={[Function]}
        type="button"
      >
        Return
      </button>
    </div>
    <div
      className="box__card box__card--tab"
    >
      <form
        className="form"
        onSubmit={[Function]}
      >
        <div
          className="form__field form__field--text"
        >
          <label
            className="form__label"
            htmlFor="origin"
          >
            Origin City
          </label>
          <div
            className="form__field--contents"
          >
            <div
              className="react-autosuggest__container"
            >
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-expanded={false}
                aria-haspopup={false}
                aria-owns="react-autowhatever-1"
                autoComplete="off"
                className="react-autosuggest__input"
                id="origin"
                name="origin"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Enter Origin City"
                role="combobox"
                type="text"
                value=""
              />
              <div
                className="react-autosuggest__suggestions-container"
                id="react-autowhatever-1"
              />
            </div>
          </div>
        </div>
        <div
          className="form__field form__field--text"
        >
          <label
            className="form__label"
            htmlFor="destination"
          >
            Destination City
          </label>
          <div
            className="form__field--contents"
          >
            <div
              className="react-autosuggest__container"
            >
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-expanded={false}
                aria-haspopup={false}
                aria-owns="react-autowhatever-1"
                autoComplete="off"
                className="react-autosuggest__input"
                id="destination"
                name="destination"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Enter Destination City"
                role="combobox"
                type="text"
                value=""
              />
              <div
                className="react-autosuggest__suggestions-container"
                id="react-autowhatever-1"
              />
            </div>
          </div>
        </div>
        <div
          className="date__picker"
        >
          <div
            className="form__field form__field--date"
          >
            <label
              className="form__label"
              htmlFor="departureDate"
            >
              Departure
            </label>
            <div
              className="form__field--contents"
            >
              <input
                defaultValue="2017-10-16"
                id="departureDate"
                min="2017-10-16"
                name="departureDate"
                type="date"
              />
            </div>
          </div>
        </div>
        <div
          className="form__field form__field--select"
        >
          <label
            className="form__label"
            htmlFor="passengers"
          >
            Passangers
          </label>
          <div
            className="form__field--contents"
          >
            <select
              id="passangers"
              name="passengers"
            >
              <option
                value="1"
              >
                1
              </option>
              <option
                value="2"
              >
                2
              </option>
              <option
                value="3"
              >
                3
              </option>
              <option
                value="4"
              >
                4
              </option>
              <option
                value="5"
              >
                5
              </option>
            </select>
          </div>
        </div>
        <div
          className="form__btn--pannel"
        >
          <button>
            Search
          </button>
        </div>
      </form>
    </div>
  </div>
  <div
    className="box"
  >
    <div
      className="box__card"
    >
      <h2>
        Refine Flight Search
      </h2>
      <div
        className="range__slider"
      >
        <div
          aria-disabled={false}
          className="input-range"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
        >
          <span
            className="input-range__label input-range__label--min"
          >
            <span
              className="input-range__label-container"
            >
              0
            </span>
          </span>
          <div
            className="input-range__track input-range__track--background"
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="input-range__track input-range__track--active"
              style={
                Object {
                  "left": "0%",
                  "width": "100%",
                }
              }
            />
            <span
              className="input-range__slider-container"
              style={
                Object {
                  "left": "0%",
                  "position": "absolute",
                }
              }
            >
              <span
                className="input-range__label input-range__label--value"
              >
                <span
                  className="input-range__label-container"
                >
                  0
                </span>
              </span>
              <div
                aria-controls={undefined}
                aria-labelledby={undefined}
                aria-valuemax={10000}
                aria-valuemin={0}
                aria-valuenow={0}
                className="input-range__slider"
                draggable="false"
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                role="slider"
                tabIndex="0"
              />
            </span>
            <span
              className="input-range__slider-container"
              style={
                Object {
                  "left": "100%",
                  "position": "absolute",
                }
              }
            >
              <span
                className="input-range__label input-range__label--value"
              >
                <span
                  className="input-range__label-container"
                >
                  10000
                </span>
              </span>
              <div
                aria-controls={undefined}
                aria-labelledby={undefined}
                aria-valuemax={10000}
                aria-valuemin={0}
                aria-valuenow={10000}
                className="input-range__slider"
                draggable="false"
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                role="slider"
                tabIndex="0"
              />
            </span>
          </div>
          <span
            className="input-range__label input-range__label--max"
          >
            <span
              className="input-range__label-container"
            >
              10000
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
